mkosi/__main__.py

```diff
@@ -25,6 +25,7 @@ def propagate_failed_return() -> Iterator[None]:
 
 @propagate_failed_return()
 def main() -> None:
+    sys.excepthook = excepthook
```

Comment:
Justification?

Reply:
This PR removes the `bin/mkosi` script in favour of using the entrypoint
```
[project.scripts]
mkosi = "mkosi.__main__:main"
```
in `pyproject.toml`, because the rootless builds make the dance we did in `bin/mkosi` if not unnecessary at least less so (see the open questions). The same would be possible with `setup.py`.

The entrypoint only takes a single function, so I moved the setting of the excepthook here, since otherwise it would only be called if the module were invoked.